### DISCUSSION TITLE
The threefold path

### DISCUSSION THREAD

**Discussion by LucaGhiringhelli** [FAIRmat-NFDI/fairmat-taxonomy#10] [General]
We are developing 3 things at once, with different teams:

1. The nomad-materials-properties plugin. This will contain a set of base sections in Metainfo representing materials properties. 
- They will be reusable in code parsers and custom schemas for characterization, synthesis, and analysis and will provide functionality for normalization into the overall nomad `results.summary` section. 
- The materials properties sections will inherit from an abstract MaterialsProperty (base-?) section in Metainfo. 
- This (Python) class is already an abstraction/selection from the several schemas underneath (present and future ones). 
- The set of materials properties sections would be a flat list for practical purposes and might contain attributes to define their classification. Typically, these classes will contain more info than the taxonomy (see below). 
- This will be developed by a selected team of domain experts and the AreaD team (Lauri, etc.). The connection between this and the taxonomy is discussed below.

2. the "NOMAD Materials Property Taxonomy" (NMPX - placeholder abbreviation, pls suggest a catchy one!). 
- This aims at building a structured taxonomy, in practice adopting owl as language, a taxonomy which will be as deep as needed (but not more). Also, the "leaf" properties might (and will) have multiple parents in the hierarchy. 
- The mapping to MaterialsProperty in Metainfo will be (necessarily?) manual, but some kind of "standard name" for each leaf property will be the same in both structures. 
- Being a manual activity, the operation will not take place at each update in Metainfo (they are typically faster), but each update in NMPX must be synced to some version of Metainfo. 
- The items in the taxonomy need to have good (=agreed upon) names, and descriptions, but also dimension and shape (currently missing). It does not contain info on the reference to methods and systems. This is done at the Metainfo level, but, again there will be a 1to1 correspondence between the leaves in NMPX and the classes in MaterialsProperty in Metainfo.
- This development is / will be coordinated by Luca, Sascha (and Chema) for the actual implementation, and trying to involve as many PIs as possible. 

3. the "Properties" menu in the GUI. 
- This is necessarily a tree. Ideally, it contains all "leaves" as NMPX but each property must have only one parent (alternatively, a property may appear in multiple menus? Let's hear the PIs).  
- This is also a manual activity (coordinated by D1, i.e., again Luca), but essentially it can be semi-automatic by just adding a "main parent" attribute to the taxonomy.

**Comment by JosePizarro3** [comment]
Hi there,

This is a very nice summary. A question regarding the connection between point 1 and 2: right now, these are two parallel definitions (as I understood from the third point in 2). Did you discuss automatizing point 2 to 1, such that we only need to work on defining new properties (like ElectronicProperty) or sub-properties (like BandGapElectronic) at the NMPX level? Or we don't want to do so?

I have the feeling that maintaining both things at the same time might end up being similar to `run` and `results`, a bit complicated on the long term, whether if the mapping is done automatically and only in one side, it will be easier and takes the burden out of having to maintain two definitions.
